cpp/include/cudf/io/text/superstate.hpp

```diff
@@ -0,0 +1,137 @@
+#pragma once
+
+#include <cmath>
+#include <cstdint>
+#include <type_traits>
+
+namespace {
+
+constexpr unsigned floorlog2(unsigned x) { return x == 1 ? 0 : 1 + floorlog2(x >> 1); }
```

Comment:
Just curious about this function. I believe the recursion will be infinite if `x==0`. Otherwise, I think the result will always be 0 since any `x > 1` value will continue the recursion until `x >> 1 == 1`.